### ISSUE TITLE
Implement automated (regression) testing

### ISSUE THREAD

**Issue by boulund** [ctmrbio/BACTpipe#46]
As a stretch-goal for the next main version of the workflow, I'd like us to implement automated regression testing. It's not very difficult, and will improve our ability to catch small bugs, typos, etc. in pull requests and commits as early as possible. 

In order to implement this in a useful way, we need at least one test case for each major path through the workflow. E.g. one gram positive, one gram negative, one with reference proteins, one which is contaminated, etc., so we can validate that all common paths through the workflow works as they should. 

I'm labeling this issue as low priority and assigning it to the BACTpipe v3.0 milestone for now.  There's no rush to implement this at this time.

**Comment by thorellk** [comment]
This is something that @abhi18av also mentioned. Maybe this is the time to implement it?

**Comment by boulund** [comment]
I absolutely think so! Not sure where the nextflow field is at now, how do people normally perform automated testing of their workflows, do you know @abhi18av ? 

I guess we need at least a minimal test data set that assembles and can be annotated, but using a minimum amount of resources to do so, so the test can be run relatively fast on a small vm?

**Comment by abhi18av** [comment]
I do agree on this as well :) 

With the new modular design of the code base, it should be easier to test at a `module` level as well as `(sub)-workflow` level.  `Github actions` seems to be the way forward, but only after the community has figured out an optimal automated testing workflow.


This week itself, there has been major progress, with https://github.com/nf-core/modules/pull/80 and https://github.com/nf-core/modules/pull/85  and in a week or so it would be stabilized. Then we can see whether we can re-use that pattern or do we have to create our own :)
